packages/getting-started/src/browser/getting-started-contribution.ts

```diff
@@ -49,11 +53,14 @@ export class GettingStartedContribution extends AbstractViewContribution<Getting
     }
 
     async onStart(app: FrontendApplication): Promise<void> {
-        if (!this.workspaceService.opened) {
-            this.stateService.reachedState('ready').then(
-                () => this.openView({ reveal: true, activate: true })
-            );
-        }
+        this.stateService.reachedState('ready').then(
+            () => this.preferenceService.ready.then(() => {
+                const showWelcomePage: boolean = this.preferenceService.get(WelcomePagePreferences.alwaysShowWelcomePage, true);
```

Comment:
I believe we still need to respect the `this.workspaceService.opened` state to only display the welcome page on start and no workspace being opened previously. I believe this is the behavior present in VS Code, but to be confirmed.